README.md

```diff
@@ -1,415 +1,86 @@
-* `OP_VAULT`: a vault UUID. Used as default vault in the `LoadStruct`, `LoadStructFromItemByTitle` and `LoadStructFromItem` functions, for all fields where the `opvault` tag is not set.
-
-### Errors
-All errors returned by Connect API are unmarshalled into a `onepassword.Error` struct:
-```go
-type Error struct {
-    StatusCode int    `json:"status"`
-    Message    string `json:"message"`
-}
-```
-
-Details of the errors can be accessed by using `errors.As()`:
-```go
-_, err := client.GetVaults()
-if err != nil{
-    var opErr *onepassword.Error
-    if errors.As(err, &opErr){
-        fmt.Printf("message=%s, status code=%d\n",
-            opErr.Message,
-            opErr.StatusCode,
-        )
-    }
-}
-```
-## Development
-
-### Building
-
-To build all packages:
-
-```sh
-make build
-```
-
-### Running Tests
-
-Run all tests:
-
-```sh
-make test
-```
-
-Run all tests with code coverage:
-
-```sh
-make test/coverage
-```
-
-## Security
+<!-- Image sourced from https://blog.1password.com/introducing-secrets-automation/ -->
+<img alt="" role="img" src="https://blog.1password.com/posts/2021/secrets-automation-launch/header.svg"/>
+
+<div align="center">
+  <h1>1Password Connect SDK for Go</h1>
+  <p>Access your 1Password items in your Go applications through your self-hosted <a href="https://developer.1password.com/docs/connect">1Password Connect server</a>.</p>
+  <a href="#✨-quickstart">
+    <img alt="Get started" src="https://user-images.githubusercontent.com/45081667/226940040-16d3684b-60f4-4d95-adb2-5757a8f1bc15.png" height="37"/>
+  </a>
+</div>
+
+---
+
+The 1Password Connect Go SDK provides access to the [1Password Connect](https://developer.1password.com/docs/connect) API, to facilitate communication with the Connect server hosted on your infrastructure and 1Password. The library is intended to be used by your applications, pipelines, and other automations to simplify accessing items stored in your 1Password vaults.
+
+## ✨ Quickstart
+
+1. Download and install the 1Password Connect Go SDK:
+   
+   ```sh
+   go get github.com/1Password/connect-sdk-go
+   ```
+
+2. Export the `OP_CONNECT_HOST` and `OP_CONNECT_TOKEN` environment variables:
+
+   ```sh
+   export OP_CONNECT_HOST=<your-connect-host> && \
+       export OP_CONNECT_TOKEN=<your-connect-token>
+   ```
+
+3. Use it in your code
+   - Read a secret:
+   
+     ```go
+     import "github.com/1Password/connect-sdk-go/connect"
+
+     func main () {
+		 client := connect.NewClientFromEnvironment()
+	 	 item, err := client.GetItem("<item-uuid>", "<vault-uuid>")
+	 	 if err != nil {
+			 log.Fatal(err)
+		 }
+	 }
+	 ```
+
+   - Write a secret:
+
+     ```go
+     import (
+         "github.com/1Password/connect-sdk-go/connect"
+	     "github.com/1Password/connect-sdk-go/onepassword"
+     )
+
+     func main () {
+	     client := connect.NewClientFromEnvironment()
+	     item := &onepassword.Item{
+			 Title:    "Secret String",
+			 Category: onepassword.Login,
```

Comment:
QQ: are the go docs populated from the repo? If not we may want to consider keeping all the usage (besides the quickstart) as part of the go docs